pkg/minikube/cluster/cluster.go

```diff
@@ -116,11 +116,19 @@ func StopHost(api libmachine.API) error {
 }
 
 // DeleteHost deletes the host VM.
+// If RHEL VM then unregister first from RHN
 func DeleteHost(api libmachine.API) error {
 	host, err := api.Load(constants.MachineName)
 	if err != nil {
 		return err
 	}
+	_, err = host.RunSSHCommand("sudo subscription-manager version")
```

Comment:
So really to make this more generic, one would let the provisoner "register" a callback function which is stored as part of the minishift config. On _DeleteHost_ you then check the config whether a callback is registered and execute it dynamically.